tiledb/sm/query/reader.cc

```diff
@@ -1294,7 +1300,17 @@ Status Reader::copy_partitioned_var_cells(
       auto var_dest = buffer_var + var_offset;
 
       // Copy offset
-      std::memcpy(offset_dest, &var_offset, offset_size);
+      if (cfg_offsets_format == "bytes") {
+        std::memcpy(offset_dest, &var_offset, offset_size);
+      } else {
+        assert(cfg_offsets_format == "elements");
+        if (fill_value_size == 1) {
```

Comment:
I know that I suggested this logic, but I am now thinking that it should be quicker to always compute `element_offset` rather than introducing a branch here. Let's remove this if-statement.